include/flatbuffers/minireflect.h

```diff
@@ -252,12 +254,20 @@ inline void IterateObject(const uint8_t *obj, const TypeTable *type_table,
     visitor->Field(i, set_idx, type, is_vector, ref, name, val);
     if (val) {
       set_idx++;
-      if (is_vector) {
-        val += ReadScalar<uoffset_t>(val);
-        auto vec = reinterpret_cast<const Vector<uint8_t> *>(val);
+      if (is_vector || is_array) {
+        auto elem_ptr = val;
+        size_t size = 0;
+        if (is_vector) {
```

Comment:
`is_vector` can be replaced with `type_table->st == ST_TABLE` if we merge array & vector?